modo/helpers.py

```diff
@@ -186,6 +186,15 @@ def is_uri(text: str):
         return False
 
 
+def split_s3_url(s3_url: str) -> tuple[str, str]:
```

Comment:
**note:** this does indeed feel dangerous indeed if we keep that, I'd start by checking if the format matches our assumption (if possible) and raise an error if it doesn't.